### PR TITLE
feat: add `--payout-address`

### PR DESCRIPTION
**Pull request by emizzle**
Allows SPs to be paid out to a separate address, keeping their profits secure. Supports https://github.com/codex-storage/codex-contracts-eth/pull/144 in the nim-codex client.

### CODE REVIEW

**Review by AuHau — APPROVED** [review]
LGTM! Thanks and good work 💪

File: tests/contracts/testMarket.nim [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
+import std/importutils
 import pkg/chronos
+import pkg/ethers/erc20
 import codex/contracts
 import ../ethertest
 import ./examples
 import ./time
 import ./deployment
 
+privateAccess(OnChainMarket) # enable access to private fields
```
Comment (markspanbroek) [inline tests/contracts/testMarket.nim:11]: 🟠 It doesn't look like you need access to private fields of `OnChainMarket` in this test? Am I missing something?

**Review by markspanbroek — APPROVED** [review]
:+1:

File: tests/contracts/testMarket.nim (hunk 1 above)
Comment (emizzle) [reply tests/contracts/testMarket.nim:11]: Good eye, Mark! `market.rewardRecipient` is used in `switchAccount`, which is not an exported field.

**Pull request merged**